### PR TITLE
Move auth_token_exp_timeout out of RBAC configuration

### LINKED ISSUE #5044: Move auth_token_exp_timeout out of RBAC configuration.
## Description

Currently, the auth_token_exp_timeout configuration option is located inside the RBAC section in api.yaml, as you can see here:

```YAML
# RBAC configuration
 rbac:
   mode: white
   auth_token_exp_timeout: 36000
```
However, the `jwt` token does not belong to RBAC, so it is better to move it out like any other configuration.

```YAML
# JWT token expiration time (in seconds)
 auth_token_exp_timeout: 36000

# RBAC configuration
 rbac:
   mode: white
```
It may be used in more places than mentioned in the task list. Make sure to change it everywhere.

## Tasks

- [x] api.yaml
- [x] api/api/configuration.py
- [x] api/api/authentication.py
- [x] api/api/spec/spec.yaml
- [x] api/scripts/migration.py
- [x] tavern tests
- [x] API configuration integration tests inside Wazuh-QA ([685](https://github.com/wazuh/wazuh-qa/pull/685))

Kind regards,
Selu.

### PR DESCRIPTION
**Pull request by vicferpoy**
Hello team, this closes #5044 .

In addition to the changes the issue requested, I unified the default value to `36000` for testing. This must be changed to a lower one when it is ready for production.

# Tests performed 
## Manager

```
============================= test session starts ==============================
platform linux -- Python 3.8.2, pytest-5.4.1, py-1.8.1, pluggy-0.13.1 -- /usr/bin/python3
cachedir: .pytest_cache
rootdir: /home/vicferpoy/Desktop/Git/wazuh/api/test/integration, inifile: pytest.ini
plugins: tavern-1.0.0
collecting ... collected 35 items

test_manager_endpoints.tavern.yaml::GET /manager/status PASSED           [  2%]
test_manager_endpoints.tavern.yaml::GET /manager/info PASSED             [  5%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[auth] PASSED [  8%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[cis-cat] PASSED [ 11%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[cluster] PASSED [ 14%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[command] PASSED [ 17%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[global] PASSED [ 20%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[localfile] PASSED [ 22%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[open-scap] PASSED [ 25%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[osquery] PASSED [ 28%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[remote] PASSED [ 31%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[rootcheck] PASSED [ 34%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[ruleset] PASSED [ 37%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[syscheck] PASSED [ 40%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[syscollector] PASSED [ 42%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[vulnerability-detector] PASSED [ 45%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration?{section}[sca] PASSED [ 48%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration PASSED    [ 51%]
test_manager_endpoints.tavern.yaml::GET /manager/stats PASSED            [ 54%]
test_manager_endpoints.tavern.yaml::GET /manager/stats/hourly PASSED     [ 57%]
test_manager_endpoints.tavern.yaml::GET /manager/stats/weekly PASSED     [ 60%]
test_manager_endpoints.tavern.yaml::GET /manager/stats/analysisd PASSED  [ 62%]
test_manager_endpoints.tavern.yaml::GET /manager/stats/remoted PASSED    [ 65%]
test_manager_endpoints.tavern.yaml::GET /manager/logs PASSED             [ 68%]
test_manager_endpoints.tavern.yaml::GET /manager/logs/summary PASSED     [ 71%]
test_manager_endpoints.tavern.yaml::GET /manager/files PASSED            [ 74%]
test_manager_endpoints.tavern.yaml::PUT /manager/files PASSED            [ 77%]
test_manager_endpoints.tavern.yaml::DELETE /manager/files PASSED         [ 80%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration/validation (OK) PASSED [ 82%]
test_manager_endpoints.tavern.yaml::GET /manager/validation (KO) PASSED  [ 85%]
test_manager_endpoints.tavern.yaml::GET /manager/configuration/{component}/{configuration} PASSED [ 88%]
test_manager_endpoints.tavern.yaml::GET /manager/api/config PASSED       [ 91%]
test_manager_endpoints.tavern.yaml::PUT /manager/api/config PASSED       [ 94%]
test_manager_endpoints.tavern.yaml::DELETE /manager/api/config PASSED    [ 97%]
test_manager_endpoints.tavern.yaml::PUT /manager/restart PASSED          [100%]

================= 35 passed, 38 warnings in 385.65s (0:06:25) ==================

```

## Cluster
```
============================= test session starts ==============================
platform linux -- Python 3.8.2, pytest-5.4.1, py-1.8.1, pluggy-0.13.1 -- /usr/bin/python3
cachedir: .pytest_cache
rootdir: /home/vicferpoy/Desktop/Git/wazuh/api/test/integration, inifile: pytest.ini
plugins: tavern-1.0.0
collecting ... collected 56 items

test_cluster_endpoints.tavern.yaml::GET /cluster/local/config PASSED     [  1%]
test_cluster_endpoints.tavern.yaml::GET /cluster/healthcheck PASSED      [  3%]
test_cluster_endpoints.tavern.yaml::GET /cluster/local/info PASSED       [  5%]
test_cluster_endpoints.tavern.yaml::GET /cluster/nodes PASSED            [  7%]
test_cluster_endpoints.tavern.yaml::GET /cluster/nodes[master-node] PASSED [  8%]
test_cluster_endpoints.tavern.yaml::GET /cluster/nodes[worker1] PASSED   [ 10%]
test_cluster_endpoints.tavern.yaml::GET /cluster/nodes[worker2] PASSED   [ 12%]
test_cluster_endpoints.tavern.yaml::GET /cluster/nodes select[ip] PASSED [ 14%]
test_cluster_endpoints.tavern.yaml::GET /cluster/nodes select[name] PASSED [ 16%]
test_cluster_endpoints.tavern.yaml::GET /cluster/nodes select[type] PASSED [ 17%]
test_cluster_endpoints.tavern.yaml::GET /cluster/nodes select[version] PASSED [ 19%]
test_cluster_endpoints.tavern.yaml::GET /cluster/status PASSED           [ 21%]
test_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/configuration PASSED [ 23%]
test_cluster_endpoints.tavern.yaml::GET /cluster/configuration/validation PASSED [ 25%]
test_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/configuration/{component}/{configuration} PASSED [ 26%]
test_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/files[master-node] PASSED [ 28%]
test_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/files[worker1] PASSED [ 30%]
test_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/files[worker2] PASSED [ 32%]
test_cluster_endpoints.tavern.yaml::DELETE /cluster/{node_id}/files[master-node] PASSED [ 33%]
test_cluster_endpoints.tavern.yaml::DELETE /cluster/{node_id}/files[worker1] PASSED [ 35%]
test_cluster_endpoints.tavern.yaml::DELETE /cluster/{node_id}/files[worker2] PASSED [ 37%]
test_cluster_endpoints.tavern.yaml::PUT /cluster/{node_id}/files[master-node] PASSED [ 39%]
test_cluster_endpoints.tavern.yaml::PUT /cluster/{node_id}/files[worker1] PASSED [ 41%]
test_cluster_endpoints.tavern.yaml::PUT /cluster/{node_id}/files[worker2] PASSED [ 42%]
test_cluster_endpoints.tavern.yaml::PUT /cluster/wrong_node/files PASSED [ 44%]
test_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/info[master-node] PASSED [ 46%]
test_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/info[worker1] PASSED [ 48%]
test_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/info[worker2] PASSED [ 50%]
test_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/logs[master-node] PASSED [ 51%]
test_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/logs[worker1] PASSED [ 53%]
test_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/logs[worker2] PASSED [ 55%]
test_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/logs/summary[worker1] PASSED [ 57%]
test_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/logs/summary[worker2] PASSED [ 58%]
test_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/stats[master-node] PASSED [ 60%]
test_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/stats[worker1] PASSED [ 62%]
test_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/stats[worker2] PASSED [ 64%]
test_cluster_endpoints.tavern.yaml::GET /cluster/wrong_node/stats PASSED [ 66%]
test_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/stats/analysisd[master-node] PASSED [ 67%]
test_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/stats/analysisd[worker1] PASSED [ 69%]
test_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/stats/analysisd[worker2] PASSED [ 71%]
test_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/stats/hourly[master-node] PASSED [ 73%]
test_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/stats/hourly[worker1] PASSED [ 75%]
test_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/stats/hourly[worker2] PASSED [ 76%]
test_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/stats/remoted[master-node] PASSED [ 78%]
test_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/stats/remoted[worker1] PASSED [ 80%]
test_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/stats/remoted[worker2] PASSED [ 82%]
test_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/stats/weekly[master-node] PASSED [ 83%]
test_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/stats/weekly[worker1] PASSED [ 85%]
test_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/stats/weekly[worker2] PASSED [ 87%]
test_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/status[master-node] PASSED [ 89%]
test_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/status[worker1] PASSED [ 91%]
test_cluster_endpoints.tavern.yaml::GET /cluster/{node_id}/status[worker2] PASSED [ 92%]
test_cluster_endpoints.tavern.yaml::GET /cluster/api/config PASSED       [ 94%]
test_cluster_endpoints.tavern.yaml::PUT /cluster/api/config PASSED       [ 96%]
test_cluster_endpoints.tavern.yaml::DELETE /cluster/api/config PASSED    [ 98%]
test_cluster_endpoints.tavern.yaml::PUT /cluster/restart PASSED          [100%]

================= 56 passed, 72 warnings in 131.27s (0:02:11) ==================

```

Related PR in `wazuh-qa`: https://github.com/wazuh/wazuh-qa/pull/718

Regards.

### CODE REVIEW

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** 5d41153dc2a4a55ef5b23bdbc665d15f89b4a6c1
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B1875/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B1875/ossec.log)
**Coverage:** 

```json
Lines executed:3.93% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B1875/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v4.0.0-B1875.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B1875/Analysis_dynamic_valgrind-v4.0.0-B1875.log)

File: api/api/configuration.py [hunk 1]
```diff
@@ -24,9 +24,9 @@
     "host": "0.0.0.0",
     "port": 55000,
     "behind_proxy_server": False,
+    "auth_token_exp_timeout": 36000,
```
Comment (davidjiglesias) [inline api/api/configuration.py:27]:
```suggestion
    "auth_token_exp_timeout": 900,
```

File: api/api/configuration/api.yaml [hunk 2]
```diff
@@ -8,10 +8,12 @@
 
 # Advanced configuration
 
+# JWT token expiration time (in seconds)
+# auth_token_exp_timeout: 36000
```
Comment (davidjiglesias) [inline api/api/configuration/api.yaml:12]:
```suggestion
# auth_token_exp_timeout: 900

```

File: api/api/spec/spec.yaml [hunk 3]
```diff
@@ -2221,6 +2221,12 @@ components:
           description: Set this option to "yes" in case the API is running behind a proxy server.
           type: boolean
           default: false
+        auth_token_exp_timeout:
+          description: Time in seconds until the token expires.
+          type: integer
+          format: int32
+          minimum: 30
+          example: 36000
```
Comment (davidjiglesias) [inline api/api/spec/spec.yaml:2229]:
```suggestion
          example: 900
```

File: api/api/spec/spec.yaml [hunk 4]
```diff
@@ -6851,9 +6851,9 @@ paths:
                 host: 0.0.0.0
                 port: 55000
                 behind_proxy_server: false
+                auth_token_exp_timeout: 36000
```
Comment (davidjiglesias) [inline api/api/spec/spec.yaml:6854]:
```suggestion
                auth_token_exp_timeout: 900
```

File: api/api/spec/spec.yaml [hunk 5]
```diff
@@ -8918,9 +8918,9 @@ paths:
...
```
Comment (davidjiglesias) [inline api/api/spec/spec.yaml:8921]:
```suggestion
                auth_token_exp_timeout: 900
```

File: api/scripts/migration.py [hunk 6]
```diff
@@ -101,7 +101,8 @@ def rename_old_fields(config: Dict) -> Dict:
     if 'basic_auth' in new_config:
         del new_config['basic_auth']
 
-    new_config['rbac'] = {"mode": "white", "auth_token_exp_timeout": 36000}
+    new_config['rbac'] = {"mode": "white"}
+    new_config['auth_token_exp_timeout'] = 36000
```
Comment (davidjiglesias) [inline api/scripts/migration.py:105]:
```suggestion
900
```

File: api/test/integration/env/configurations/base/manager/api.yaml [hunk 7]
```diff
@@ -6,10 +6,12 @@ behind_proxy_server: no
 
 # Advanced configuration
 
+# JWT token expiration time (in seconds)
+auth_token_exp_timeout: 36000
```
Comment (davidjiglesias) [inline api/test/integration/env/configurations/base/manager/api.yaml:10]:
```suggestion
auth_token_exp_timeout: 900
```

File: api/test/integration/test_cluster_endpoints.tavern.yaml [hunk 8]
```diff
@@ -2716,9 +2716,9 @@ stages:
       status_code: 200
       json:
         <<: *api_response
+        auth_token_exp_timeout: 36000
```
Comment (davidjiglesias) [inline api/test/integration/test_cluster_endpoints.tavern.yaml:2719]:
```suggestion
        auth_token_exp_timeout: 900
```

File: api/api/configuration.py (hunk 1 above)
Comment (vicferpoy) [reply api/api/configuration.py:27]: As this is not ready for production yet, I rather have this long timeout for testing.

**Pull request merged**